prov/verbs/src/fi_verbs.c

```diff
@@ -295,10 +295,12 @@ int vrb_get_rai_id(const char *node, const char *service, uint64_t flags,
 				(*rai)->ai_dst_addr, VERBS_RESOLVE_TIMEOUT);
 	if (ret) {
 		VERBS_INFO_ERRNO(FI_LOG_FABRIC, "rdma_resolve_addr", errno);
-		ofi_straddr_log(&vrb_prov, FI_LOG_INFO, FI_LOG_FABRIC,
-				"src addr", (*rai)->ai_src_addr);
-		ofi_straddr_log(&vrb_prov, FI_LOG_INFO, FI_LOG_FABRIC,
-				"dst addr", (*rai)->ai_dst_addr);
+		if ((*rai)->ai_src_addr)
+			ofi_straddr_log(&vrb_prov, FI_LOG_INFO, FI_LOG_FABRIC,
+					"src addr", (*rai)->ai_src_addr);
+		if ((*rai)->ai_dst_addr)
+			ofi_straddr_log(&vrb_prov, FI_LOG_INFO, FI_LOG_FABRIC,
+					"dst addr", (*rai)->ai_dst_addr);
```

Comment:
It may be better to modify ofi_straddr_log_internal() to display "(null)" when passed a null address.  That way, we avoid adding these checks, plus the logging is consistent, rather than it not being there at all.

Reply:
Good call

Reply:
Updated